src/main/java/hudson/plugins/git/GitChangeSet.java

```diff
@@ -49,6 +50,15 @@
     private static final String ISO_8601 = "yyyy-MM-dd'T'HH:mm:ss";
     private static final String ISO_8601_WITH_TZ = "yyyy-MM-dd'T'HH:mm:ssX";
 
+    // Nearly ISO dates generated by git whatchanged --format=+ci
+    // Look like '2015-09-30 08:21:24 -0600'
+    // ISO is    '2015-09-30T08:21:24-06:00'
+    private static final Pattern NEARLY_ISO_DATE = Pattern.compile("^([0-9]{4}-[0-9]{2}-[0-9]{2})" + // YYYY-MM-DD
```

Comment:
I'd rather vote for replacing the logic by an external implementation. E.g. using a parser from http://joda-time.sourceforge.net/apidocs/org/joda/time/format/ISODateTimeFormat.html